static/js/selectors/websites.ts

```diff
@@ -0,0 +1,4 @@
+// TODO: how to type state?
+
+export const getWebsite = (state: any, uuid: string) =>
```

Comment:
we might want to grab a library like [reselect](https://github.com/reduxjs/reselect) to get these selectors to be memoized and all that

Reply:
Sure but do you think it would be useful for this particular selector? In the examples they only use `createSelector` for more complicated selectors: https://github.com/reduxjs/reselect#selectorsindexjs

Reply:
Also, any thoughts on what to use instead of `any` for a type for `state`?

Reply:
I think you can use `ReturnType<typeof configureStore>` for it. We should actually create a type alias for that, like

```js
type Store = ReturnType<typeof configureStore>
```

Reply:
in the `configureStore.ts` file, and then we can import that elsewhere

Reply:
Thanks, that fixed it

Reply:
I think the advantage to using reselect is the memoization, which then in turn reduces the number of times that react will needlessly re-render (because for objects react only does a shallow compare).

for this one I think you could do something like:

```js
const getWebsite = createSelector(
  state => state.entities?.websites,
  websites => (
    lodash.memoize(name => websites[name])
  )
)
```

this is a pattern we use in open-discussions in a few places. then basically you get two things 1) the return value of the selector (the generated fucnction) will be memoized, and regenerated whenever the value in the store changes and 2) the memoized function has the websites within a closure so it will be usable within the component.

then you can do in the component:

```js
const memoizedWebsites = useSelector(getWebsite)
const website = memoizedWebsites(name)
```

one example of this pattern in open-discussions in the `learningResourceSelector` (although that one's more complicated because the accessor function needs to take two params)

Reply:
if you do the above probably good ot name the selector `getWebsiteCursor` or something that captures that it returns an accessor function instead of a straight up value, then you'd do something like

```js
const websiteCursor = useSelector(getWebsiteCursor)
const website = websiteCursor(name)
```